### ISSUE TITLE
Url rewritten in lowercase when accessing child route without redirect

### ISSUE THREAD

**Issue by SennaSanzo** [angular/angular#53859]
### Which @angular/* package(s) are the source of the bug?

core, router

### Is this a regression?

Yes

### Description

Hi everyone,

I really don't know if this is really a bug. I probably did something wrong with my routes configuration. But here is my problem:

I've got this route configuration:

```Javascript
export const routes: Routes = [    
    {
        path: "webspaces/:webSpaceName",
        title: "Web Spaces",
        children: [
            {
                path: "",
                redirectTo: "login",
                pathMatch: "full"
            },
            {
                path: 'login',
                title: 'Login',
                canActivate: [isAuthenticatedGuard],
                component: LoginComponent
            },
            {
                path: 'dashboard',
                title: 'Dashboard',
                canActivate: [isAuthenticatedGuard],
                component: DashboardComponent
            },
        ]
    },    
    //otherwise redirect to home
    { path: '**', component: HomeComponent}
];
```

As you can see I want an url structured this way : /webspaces/:webSpaceName/login or /dashboard. I used an empty child route to redirect to login component. If I type /webspaces/**WS1** everything works correctly an I'm redirected to the login component and once logged in to dashboard component without any issue. But if I try to type the url directly in the browser like so : /webspaces/**WS1**/login, the url is automatically rewritten in lowercase as /webspaces/**ws1**/login which can be problematic on my project where the parameter is case sensitive.

For context, I'm currently only on local dev environment accessing the app by localhost.

### Please provide the exception or error you saw

```true
A children url is rewritten as lowercase when accessed directly by typing it in the browser navigation bar.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.0.7
Node: 20.9.0
Package Manager: npm 10.1.0
OS: win32 x64

Angular: 17.0.7
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.7
@angular-devkit/build-angular   17.0.7
@angular-devkit/core            17.0.7
@angular-devkit/schematics      17.0.7
@schematics/angular             17.0.7
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.2
```

**Comment by atscott** [comment]
I'm sorry but we can't reproduce the problem following the instructions you provided.

If the problem still exists in your application please [open a new issue](https://github.com/angular/angular/issues/new/choose) and follow the instructions in the issue template.

I attempted to reproduce this but the redirect did not change the URL to lowercase: https://stackblitz.com/edit/stackblitz-starters-yidfad

**Comment by SennaSanzo** [comment]
After seeing your response I tried to dig more this problem by creating a completely new angular app from scratch by using Angular CLI and only the router configuration I mentioned above (+ 2 test components). And what I found is very strange : I still got erratic lowercase url rewriting but I started to suspect Google Chrome and not Angular. Since I started to test on FireFox, the problem disappeared.

I found an old stackoverflow message related to this issue :

[Chrome changes URL case](https://stackoverflow.com/questions/50298340/chrome-changes-url-case)

I exactly encounter the same behavior : It appears the problem comes from the Chrome url history. If you type in the navigation bar the same url but with different case, Chrome will rewrite it with the case of the previous history entry.

Tell me what can I do ?
